File: src/main/java/org/moinex/repository/wallettransaction/WalletTypeRepository.java
/*
 * Filename: WalletTypeRepository.java
 * Created on: September 29, 2024
 * Author: Lucas Araújo <araujolucas@dcc.ufmg.br>
 */

package org.moinex.repository.wallettransaction;

import java.util.List;
import java.util.Optional;
import org.moinex.model.wallettransaction.WalletType;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface WalletTypeRepository extends JpaRepository<WalletType, Integer> {

    /**
     * Find all wallet types ordered by name ascending
     * @return List of wallet types
     */
    List<WalletType> findAllByOrderByNameAsc();

    /**
     * Find a wallet type by its name
     * @param name The name of the wallet type
     * @return The wallet type
     */
    Optional<WalletType> findByName(String name);

    /**
     * Check if a wallet type with the given name exists
     * @param name The name of the wallet type
     * @return True if the wallet type exists, false otherwise
     */
    boolean existsByName(String name);
}

File: src/main/kotlin/org/moinex/repository/wallettransaction/WalletTypeRepository.kt
/*
 * Filename: WalletTypeRepository.kt (original filename: WalletTypeRepository.java)
 * Created on: September 29, 2024
 * Author: Lucas Araújo <araujolucas@dcc.ufmg.br>
 *
 * Migrate to Kotlin on 03/03/2026
 */

package org.moinex.repository.wallettransaction

import org.moinex.model.wallettransaction.WalletType
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface WalletTypeRepository : JpaRepository<WalletType, Int> {
    fun findAllByOrderByNameAsc(): List<WalletType>

    fun findByName(name: String): Optional<WalletType>

    fun existsByName(name: String): Boolean
}
